OpenSourcePolitics/decidim-module-half_sign_up
# feat:validates_phone_number_fr

🎩 Description
Add phone *validation format when user use quick_auth sms.

*The phone number must adhere to the national format:
- The second digit must be either 6 or 7: 06 XX XX XX XX / 07 XX XX XX XX /6 XX XX XX XX / 7 XX XX XX XX
- The format must contain exactly 10 digits.

📌 Related Issues
[Notion card](https://www.notion.so/opensourcepolitics/Toulouse-Validate-phone-number-format-27912ddb01d94c3ba359cc495a336365?pvs=4)

**Testing**

1. As admin go to dashboard
2. Go to "authentication_settings" and verify that "Enable partial sign-up and sign-in using SMS verification" is checked.
3. Navigate to Processes and select one process.
4. Access the budget section and click on settings.
5. Uncheck "Vote based on ZIP code: allows participants to vote on budgets matching their entered ZIP code."
6. Return to the landing page and select the previously chosen process.
7. Go to the budget section and click on "start voting" (you'll be directed to the quick_auth sms form 🎉).
8. Test by changing the phone_country AND filling in the phone number (valid and invalid).
Expectations:
- When FR is selected, the button should be disabled.
- If an incorrect phone format is entered, the error message should appear after tapping the last digit (9 or 10).
- The button should be enabled when a valid phone format is entered.

📷 Screenshots
<img width="897" alt="Capture d’écran 2024-06-06 à 19 26 00" src="https://github.com/OpenSourcePolitics/decidim-module-half_sign_up/assets/143180473/2097893d-4dec-4a20-b91b-80bdbfeb12cd">
<img width="923" alt="Capture d’écran 2024-06-06 à 19 26 59" src="https://github.com/OpenSourcePolitics/decidim-module-half_sign_up/assets/143180473/daf2ae54-1245-4714-b7e3-37b10275fd02">
